Regression: error compiling the system

Did a git pull on 2.3.2018 and suddenly make fails:

```
$ make
cd src; make
make[1]: Entering directory '/home/rz/linux-prog/nit-lang/nit/src'
***************************************************************
* Compile binaries from NIT source files                      *
***************************************************************
./git-gen-version.sh
test -d ../bin || mkdir ../bin
./nitc_0 --semi-global -v --dir ../bin nitc.nit nitpick.nit nit.nit nitdoc.nit nitls.nit nitunit.nit picnit.nit nitpretty.nit nitmetrics.nit nitx.nit nitlight.nit nitserial.nit nitrestful.nit
*** PARSE ***
*** SEMANTIC ANALYSIS ***
../lib/curl/curl.nit:296,6--32: Error: untyped attribute `curl$CurlMail$supported_outgoing_protocol`. Implicit typing allowed only for literals and new.
                var supported_outgoing_protocol = ["smtp", "smtps"]
                    ^
Errors: 1. Warnings: 0.
Makefile:42: recipe for target 'bin/nitc' failed
make[1]: *** [bin/nitc] Error 1
make[1]: Leaving directory '/home/rz/linux-prog/nit-lang/nit/src'
Makefile:43: recipe for target 'tools' failed
make: *** [tools] Error 2
```

--- a/Makefile
+++ b/Makefile
@@ -29,10 +29,13 @@ all: tools man
 	@echo "To configure your shell environment, execute the following command:"
 	@echo "    source misc/nit_env.sh install"
 
+more:
+	cd src; make more
+
 # Compile all programs in `contrib`, `examples` and `src`.
 #
 # Furthermore, build the toolchain’s `man` pages.
-full: all
+full: all more
 	for directory in $(extras); do \
 		(cd "$${directory}" && $(MAKE)) || exit 1; \
 	done

--- a/src/Makefile
+++ b/src/Makefile
@@ -16,15 +16,20 @@
 
 NITCOPT=--semi-global
 OLDNITCOPT=--semi-global
-OBJS=nitc nitpick nit nitdoc nitls nitunit picnit nitpretty nitmetrics nitx nitlight nitserial nitrestful
+OBJS=nitc nitpick nit nitls nitunit picnit nitx nitlight nitserial nitrestful
 SRCS=$(patsubst %,%.nit,$(OBJS))
 BINS=$(patsubst %,../bin/%,$(OBJS))
+MOREOBJS=nitdoc nitweb nitcatalog nitmetrics nitpretty nitweb
+MORESRCS=$(patsubst %,%.nit,$(MOREOBJS))
+MOREBINS=$(patsubst %,../bin/%,$(MOREOBJS))
 DEPS=$(wildcard *.nit */*.nit) parser/parser.nit
 
 # Workaround: Cygwin requires peflags to bootstrap
 peflags := ${shell which peflags 2>/dev/null}
 
 all: ../bin/nitc
+
+more: ../bin/nitdoc
 
 pre-build:
 	./git-gen-version.sh
@@ -46,6 +51,9 @@ endif
 	./git-gen-version.sh
 	test -d ../bin || mkdir ../bin
 	./nitc_0 ${NITCOPT} -v --dir ../bin $(SRCS)
+
+../bin/nitdoc: ../bin/nitc $(DEPS)
+	../bin/nitc ${NITCOPT} -v --dir ../bin $(MORESRCS)
 
 $(OBJS): nitc_0 $(DEPS)
 	./git-gen-version.sh

--- a/src/Makefile
+++ b/src/Makefile
@@ -23,7 +23,7 @@ BINS=$(patsubst %,../bin/%,$(OBJS))
 # Workaround: Cygwin requires peflags to bootstrap
 peflags := ${shell which peflags 2>/dev/null}
 
-all: bin/nitc
+all: ../bin/nitc
 
 pre-build:
 	./git-gen-version.sh
@@ -38,7 +38,7 @@ ifdef peflags
 	${peflags} --cygwin-heap=2048 nitc_0
 endif
 
-bin/nitc: nitc_0 parser/parser.nit
+../bin/nitc: nitc_0 parser/parser.nit
 	@echo '***************************************************************'
 	@echo '* Compile binaries from NIT source files                      *'
 	@echo '***************************************************************'

--- a/src/Makefile
+++ b/src/Makefile
@@ -19,6 +19,7 @@ OLDNITCOPT=--semi-global
 OBJS=nitc nitpick nit nitdoc nitls nitunit picnit nitpretty nitmetrics nitx nitlight nitserial nitrestful
 SRCS=$(patsubst %,%.nit,$(OBJS))
 BINS=$(patsubst %,../bin/%,$(OBJS))
+DEPS=$(wildcard *.nit */*.nit) parser/parser.nit
 
 # Workaround: Cygwin requires peflags to bootstrap
 peflags := ${shell which peflags 2>/dev/null}
@@ -28,7 +29,7 @@ all: ../bin/nitc
 pre-build:
 	./git-gen-version.sh
 
-nitc_0: ../c_src/nitc parser/parser.nit
+nitc_0: ../c_src/nitc $(DEPS)
 	@echo '***************************************************************'
 	@echo '* Compile nitc_0 from NIT source files                        *'
 	@echo '***************************************************************'
@@ -38,7 +39,7 @@ ifdef peflags
 	${peflags} --cygwin-heap=2048 nitc_0
 endif
 
-../bin/nitc: nitc_0 parser/parser.nit
+../bin/nitc: nitc_0 $(DEPS)
 	@echo '***************************************************************'
 	@echo '* Compile binaries from NIT source files                      *'
 	@echo '***************************************************************'
@@ -46,7 +47,7 @@ endif
 	test -d ../bin || mkdir ../bin
 	./nitc_0 ${NITCOPT} -v --dir ../bin $(SRCS)
 
-$(OBJS): nitc_0 parser/parser.nit
+$(OBJS): nitc_0 $(DEPS)
 	./git-gen-version.sh
 	./nitc_0 ${NITCOPT} -v $@.nit
 

